app/forms/questionnaire_form.py

```diff
@@ -386,11 +386,25 @@ def _get_period_limits(
         return minimum, maximum
 
     def _get_formatted_calculation_values(
-        self, answers_list: Sequence[str]
+        self, answers_sequence: Sequence[str]
     ) -> list[str]:
+        answers_list = list(answers_sequence)
+        block_id = self.location.block_id if self.location else None  # type: ignore
+        # type ignore self.location always exists at this point
+        list_name = (
```

Comment:
Do we only want to call this method if the block has dynamic answers, so use the `self.block_has_dynamic_answer` to check if we need to make this call?

Reply:
Yeah, makes sense, no need to do this for all.